openlibrary/core/waitinglist.py

```diff
@@ -132,6 +134,13 @@ def find(cls, user_key, identifier, itemname=None):
         if result:
             return result[0]
 
+    @classmethod
+    @cache.memoize(
+        engine="memcache", key="gt-waiting-loan", expires=5 * dateutil.MINUTE_SECS
```

Comment:
Here we are going to want to store a unique value for each individual patron's loans, and therefore the cache key is going to have to be based on the user_key (which won't exist in the decorator declaration. We therefore need the key to be a lambda function, e.g.

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/models.py#L105